src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -67,19 +67,29 @@ Copyright (c) .NET Foundation. All rights reserved.
     if the output semaphore file is out of date with respect to the inputs.
     ============================================================
     -->
-   <UsingTask TaskName="ILLink" AssemblyFile="$(ILLinkTasksAssembly)" />
-   <Target Name="_RunILLink"
-           DependsOnTargets="_ComputeManagedAssembliesToLink"
-           Inputs="$(MSBuildAllProjects);@(_ManagedAssembliesToLink);@(TrimmerRootDescriptor);@(ReferencePath)"
-           Outputs="$(_LinkSemaphore)">
-
-     <Delete Files="@(_LinkedResolvedFileToPublishCandidates)" />
-     <ILLink AssemblyPaths="@(_ManagedAssembliesToLink)"
-             ReferenceAssemblyPaths="@(ReferencePath)"
-             RootAssemblyNames="@(IntermediateAssembly->'%(Filename)')"
-             RootDescriptorFiles="@(TrimmerRootDescriptor)"
-             OutputDirectory="$(IntermediateLinkDir)"
-             ExtraArgs="-u copyused -c copyused -l none --skip-unresolved true $(_ExtraTrimmerArgs)" />
+  <UsingTask TaskName="ILLink" AssemblyFile="$(ILLinkTasksAssembly)" />
+  <Target Name="_RunILLink"
+         DependsOnTargets="_ComputeManagedAssembliesToLink"
+          Inputs="$(MSBuildAllProjects);@(_ManagedAssembliesToLink);@(TrimmerRootDescriptor);@(ReferencePath)"
+          Outputs="$(_LinkSemaphore)">
+
+    <!-- When running from Desktop MSBuild, DOTNET_HOST_PATH is not set.
+         In this case, explicitly specify the path to the dotnet host. -->
+    <PropertyGroup Condition=" '$(DOTNET_HOST_PATH)' == '' ">
+      <_DotNetHostDirectory>$(NetCoreRoot)</_DotNetHostDirectory>
```

Comment:
I'm curious is the passing now because this is empty and it ends up using dotnet.exe on PATH? Or did we already get a stage 0 with $(NETCoreRoot)?

Reply:
We have it in the stage 0. I don't think any logic related to this task would actually find dotnet on the path (the task uses the inputs or DOTNET_HOST_PATH directly). This PR was failing on windows when I introduced https://github.com/dotnet/sdk/pull/3192/files#diff-a532d5eabf1063f35b5fda0af8edd118R129, but passes again with the fix.